Sources/XcodeProj/Objects/Configuration/BuildSettings.swift

```diff
@@ -1,4 +1,82 @@
 import Foundation
 
 /// Build settings.
-public typealias BuildSettings = [String: Any]
+public typealias BuildSettings = [String: BuildSetting]
+
+private let yes = "YES"
+private let no = "NO"
+
+public enum BuildSetting: Sendable, Equatable {
+    case string(String)
+    case array([String])
+
+    public var stringValue: String? {
+        if case let .string(value) = self {
+            value
+        } else {
+            nil
+        }
+    }
+
+    public var boolValue: Bool? {
+        if case let .string(value) = self {
+            switch value {
+            case yes: true
+            case no: false
+            default: nil
```

Comment:
Xcode's behavior is actually defaulting to `false` if the value is defined but it's not `YES` or `NO`. But conceptually, this feels better, so I'll let you decide what's better here 😅

Reply:
Makes sense!